File: systems/drone_router.py
class DroneRouter:
    def __init__(self):
        self.drones = [
            {"id": "D-01", "priority": "CRITICAL", "status": "MISROUTED"},
            {"id": "D-02", "priority": "STANDARD", "status": "MISROUTED"},
            {"id": "D-03", "priority": "CRITICAL", "status": "MISROUTED"},
            {"id": "D-04", "priority": "LOW",      "status": "MISROUTED"},
            {"id": "D-05", "priority": "STANDARD", "status": "MISROUTED"},
            {"id": "D-06", "priority": "CRITICAL", "status": "MISROUTED"},
            {"id": "D-07", "priority": "LOW",      "status": "MISROUTED"},
            {"id": "D-08", "priority": "STANDARD", "status": "MISROUTED"},
        ]

    def scan_drones(self):
        print("  ┌─────────┬───────────┬────────────┐")
        print("  │ Drone   │ Priority  │ Status     │")
        print("  ├─────────┼───────────┼────────────┤")
        for d in self.drones:
            print(f"  │ {d['id']:<7s} │ {d['priority']:<9s} │ {d['status']:<10s} │")
        print("  └─────────┴───────────┴────────────┘")

    def reroute_next(self):
        for d in self.drones:
            if d["status"] == "MISROUTED":
                d["status"] = "CORRECTED"
                print(f"    Drone {d['id']} ({d['priority']}) rerouted → CORRECTED")
                return
        print("    All drones already corrected.")

    def is_goal_met(self):
        return all(d["status"] == "CORRECTED" for d in self.drones)

    def get_status_text(self):
        corrected = sum(1 for d in self.drones if d["status"] == "CORRECTED")
        total = len(self.drones)
        if corrected == total:
            indicator = "[OK]"
        else:
            indicator = "[!!]"

        return f"""  SECTOR 12 DRONE GRID
  Status:     {indicator} {corrected}/{total} drones corrected
"""

File: unity/neo-kyoto/Assets/Scripts/Systems/DroneRouter.cs
using System;
using System.Collections.Generic;

namespace NeoKyoto.Systems
{
    /// <summary>Sector 12's drone grid — the C2 target. Eight drones, all misrouted.</summary>
    public class DroneRouter
    {
        public class Drone
        {
            public string Id;
            public string Priority;
            public string Status;
        }

        public readonly List<Drone> Drones = new List<Drone>();

        public Action<string> Output;
        public event Action Changed;

        public DroneRouter()
        {
            Add("D-01", "CRITICAL");
            Add("D-02", "STANDARD");
            Add("D-03", "CRITICAL");
            Add("D-04", "LOW");
            Add("D-05", "STANDARD");
            Add("D-06", "CRITICAL");
            Add("D-07", "LOW");
            Add("D-08", "STANDARD");
        }

        private void Add(string id, string priority)
        {
            Drones.Add(new Drone { Id = id, Priority = priority, Status = "MISROUTED" });
        }

        private void Print(string s) { if (Output != null) Output(s); }

        public long ScanDrones()
        {
            int remaining = 0;
            foreach (var d in Drones) if (d.Status == "MISROUTED") remaining++;

            Print("  ┌─────────┬───────────┬────────────┐");
            Print("  │ Drone   │ Priority  │ Status     │");
            Print("  ├─────────┼───────────┼────────────┤");
            foreach (var d in Drones)
                Print("  │ " + d.Id.PadRight(7) + " │ " + d.Priority.PadRight(9) + " │ " + d.Status.PadRight(10) + " │");
            Print("  └─────────┴───────────┴────────────┘");
            Print("    " + remaining + " drones still misrouted.");
            return remaining;
        }

        public object RerouteNext()
        {
            foreach (var d in Drones)
            {
                if (d.Status == "MISROUTED")
                {
                    d.Status = "CORRECTED";
                    Print("    Drone " + d.Id + " (" + d.Priority + ") rerouted → CORRECTED");
                    if (Changed != null) Changed();
                    return d.Id;
                }
            }
            Print("    All drones already corrected.");
            return null;
        }

        public bool IsGoalMet()
        {
            foreach (var d in Drones) if (d.Status != "CORRECTED") return false;
            return true;
        }

        public string GetStatusText()
        {
            int corrected = 0;
            foreach (var d in Drones) if (d.Status == "CORRECTED") corrected++;
            string indicator = corrected == Drones.Count ? "[OK]" : "[!!]";
            return "  SECTOR 12 DRONE GRID\n" +
                   "  Status:     " + indicator + " " + corrected + "/" + Drones.Count + " drones corrected\n";
        }
    }
}
